src/electron/routing_service.ts

```diff
@@ -173,16 +175,37 @@ export class RoutingDaemon {
     return response;
   }
 
-  // Use #onceDisconnected to be notified when the connection terminates.
-  stop() {
+  private async writeReset() {
+    return new Promise<void>((resolve, reject) => {
+      const written = this.socket.write(JSON.stringify(
+        {action: RoutingServiceAction.RESET_ROUTING, parameters: {}} as RoutingServiceRequest),
+        (err) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve();
+          }
+        });
+      if (!written) {
+        reject(new Error("Write failed"));
+      }
+    });
+  }
+
+  async stop() {
     if (!this.socket) {
       // Never started.
       this.fulfillDisconnect();
       return;
     }
+    if (this.stopping) {
+      // Already stopped.
+      return;
+    }
+    this.stopping = true;
 
-    this.socket.write(JSON.stringify(
-        {action: RoutingServiceAction.RESET_ROUTING, parameters: {}} as RoutingServiceRequest));
+    await this.writeReset();
```

Comment:
Should we `return this.writeReset()`? The caller shoud already awaiting for `onceDisconnected` and if we return the promise, it can choose whether to await for `stop`.

Reply:
OK, done